src/Features/Core/Portable/NameTupleElement/AbstractNameTupleElementCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,120 @@
+                new MyCodeAction(
+                    string.Format(FeaturesResources.Add_tuple_element_name_0, elementName),
+                    c => AddNamedElementAsync(document, span, cancellationToken)));
+        }
+
+        private async Task<(SyntaxNode root, TArgumentSyntax argument, string argumentName)> TryGetArgumentInfo(
+            Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            if (document.Project.Solution.Workspace.Kind == WorkspaceKind.MiscellaneousFiles)
+            {
+                return default;
+            }
+
+            if (span.Length > 0)
+            {
+                return default;
+            }
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var position = span.Start;
+            var token = root.FindToken(position);
+            if (token.Span.Start == position &&
+                IsCloseParenOrComma(token))
+            {
+                token = token.GetPreviousToken();
+                if (token.Span.End != position)
+                {
+                    return default;
+                }
+            }
+
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var argument = root.FindNode(token.Span)
+                .GetAncestorsOrThis<TArgumentSyntax>()
+                .FirstOrDefault(node => syntaxFacts.IsTupleExpression(node.Parent));
+
+            if (argument == null || !syntaxFacts.IsSimpleArgument(argument))
+            {
+                return default;
+            }
+
+            var tuple = (TTupleExpressionSyntax)argument.Parent;
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var tupleType = semanticModel.GetTypeInfo(tuple, cancellationToken).ConvertedType as INamedTypeSymbol;
+            if (tupleType == null)
+            {
+                return default;
+            }
+
+            syntaxFacts.GetPartsOfTupleExpression<TArgumentSyntax>(tuple, out _, out var arguments, out _);
+            var argumentIndex = arguments.IndexOf(argument);
+            var elements = tupleType.TupleElements;
+            if (elements.IsDefaultOrEmpty || argumentIndex >= elements.Length)
+            {
+                return default;
+            }
+
+            var element = elements[argumentIndex];
+            if ((object)element == element.CorrespondingTupleField)
+            {
+                return default;
+            }
+
+            return (root, argument, element.Name);
+        }
+
+        private async Task<Document> AddNamedElementAsync(Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            var (root, argument, elementName) = await TryGetArgumentInfo(document, span, cancellationToken);
+
+            var newArgument = WithName(argument, elementName).WithTriviaFrom(argument);
```

Comment:
Technically this could screw up trivia a bit.  Specifically, if the argument has leading trivia.  Not sure how important it is to you to get that right.